ac/ac-induction/src/TestPart/Test.jsx

```diff
@@ -9,70 +10,96 @@ import Correction from './Correction';
-      data.suffisants
-    );
-    const newList = [...tmpList];
-    newList[data.indexCurrent].correction = correction;
-    dataFn.objInsert(
-      newList,
-      feedback ? 'listIndexTestWithFeedback' : 'listIndexTest'
-    );
-    if (feedback) {
-      dataFn.objInsert(true, 'feedbackOpen');
-    } else if (data.indexCurrent === nbTest - 1) {
-      dataFn.objInsert(0, 'indexCurrent');
-      dataFn.objInsert(data.indexPart + 1, 'indexPart');
-    } else dataFn.objInsert(data.indexCurrent + 1, 'indexCurrent');
-  };
+    const clickHandler = () => {
+      if (!disable) {
+        const index = tmpList[data.indexCurrent];
+        const caseAnswer = index.selectedChoice
+          ? 0
+          : index.realIndex % 2 === 0 ? 1 : 2;
+        const correction = Correction(
+          examples[index.realIndex].isIncorrect,
+          caseAnswer,
+          index.selectedProperties,
+          stringToArray(examples[index.realIndex].respectedProperties),
+          data.contradictories,
+          data.unnecessaries,
+          data.suffisants
+        );
+        const newList = [...tmpList];
+        newList[data.indexCurrent].correction = correction;
+        dataFn.objInsert(
+          newList,
+          feedback ? 'listIndexTestWithFeedback' : 'listIndexTest'
+        );
+        if (feedback) {
+          dataFn.objInsert(true, 'feedbackOpen');
+        } else if (data.indexCurrent === nbTest - 1) {
+          dataFn.objInsert(0, 'indexCurrent');
+          dataFn.objInsert(data.indexPart + 1, 'indexPart');
+        } else dataFn.objInsert(data.indexCurrent + 1, 'indexCurrent');
+      }
+    };
 
-  return (
-    <ExMain>
-      <ExContainer>
-        <ImgBis url={examples[tmpList[data.indexCurrent].realIndex].url} />
-      </ExContainer>
-      <ExContainer style={{ padding: '20px' }}>
-        <ResponsePanel
-          {...{ title, examples, properties, tmpList, feedback, data, dataFn }}
-        />
-        <ExButton className="btn btn-default" onClick={clickHandler}>
-          Submit
-        </ExButton>
-      </ExContainer>
-      {feedback && (
-        <ModalSubmit
-          examples={examples}
-          properties={properties}
-          dataFn={dataFn}
-          data={data}
-          nbTestFeedback={nbTestFeedback}
-        />
-      )}
-    </ExMain>
-  );
-};
+    return (
+      <ExMain>
+        <ExContainer>
+          <ImgBis
+            url={examples[tmpList[data.indexCurrent].realIndex].url}
+            color="black"
+          />
+        </ExContainer>
+        <ExContainer style={{ padding: '20px' }}>
+          <ResponsePanel
+            {...{
+              title,
+              examples,
+              properties,
+              tmpList,
+              feedback,
+              data,
+              dataFn,
+              setDisable
+            }}
+          />
+          <ExButton
+            className="btn btn-default"
+            onClick={clickHandler}
+            disabled={disable}
```

Comment:
You are doing quite complicated things here. I think you should just have here something like: 
`disabled={tmpList[data.indexCurrent].selectedProperties.length < 1}` 

I think that would be reactive, unlike your current implementation